### ISSUE TITLE
There is no "extension," actually.

### ISSUE THREAD

**Issue by ghost** [Keruspe/GPaste#253]
**Please remove the option**. I spent half an hour trying to figure why there was no GPaste into the top panel. (Gnome is a nightmare, most programs could just use an indicator icon - but Gnome wants you to "develop an extension" because they want to drive you insane.) Please just disable the option, or make GPaste platform agnostic and use appindicator like all normal programs.

enabled:

![image](https://user-images.githubusercontent.com/43324854/45651430-2e773000-bada-11e8-87e0-b46125a666e5.png)

nothing happens:

![image](https://user-images.githubusercontent.com/43324854/45651817-44d1bb80-badb-11e8-9e34-be554210b8b0.png)

**Comment by lhorace** [comment]
Yeah, either or would be much appreciated here, yet learn a new keyboard shortcut. I'm not a keyboard warrior myself like most people, prefers mouse clicks.

**Comment by andyholmes** [comment]
Not to but in, because I don't know what distros you fellows are using, but what's far more likely is your distribution has decided a few kilobytes is worth splitting the extension into a separate package that you haven't installed.

Kerupse, regardless of that not being your fault, maybe it's worth a simple file-exists check connected to the sensitivity property.

**Comment by esplinr** [comment]
@andyholmes is correct.

On Ubuntu 18.04 there is a separate package gnome-shell-extensions-gpaste.

The one at extensions.gnome.org is deprecated and no longer maintained.

**Comment by AlexanderMB** [comment]
The extension works, but you need to reload gnome-shell.

**Comment by Keruspe** [comment]
@AlexanderMB Note that this is true for basically all extensions

@andyholmes I already do that for the other optional stuff but extensions are a little special in this regards since they can be placed in several different locations.

a gnome-shell api telling if an extension is loaded or not would be really nice to have there

**Comment by andyholmes** [comment]
@Keruspe there is a DBus interface `org.gnome.Shell.Extensions` at `/org/gnome/Shell`. I believe you could call `GetExtensionInfo(uuid)` to return a GVariant of the extension object and check the `state` member. Not sure what the values correspond to off-hand though.

**Comment by Keruspe** [comment]
I did not know of this one, that’s exactly what I need

**Comment by Keruspe** [comment]
FTR: `gdbus call -e -d org.gnome.Shell -o /org/gnome/Shell -m org.gnome.Shell.Extensions.GetExtensionInfo '"GPaste@gnome-shell-extensions.gnome.org"'`

**Comment by artogahr** [comment]
Suggesting issue closure given inactivity and irrelevance

**Comment by Keruspe** [comment]
No, this is quite easy to fix, just had other stuff wip (which are still not fully done :/).
I also got my 2nd, 3rd and 4th kid in the meantime which reduced the amount of time I can use to work on this (hopefully it'll get better in the next months).
Also, I now work with a higher priority on issues suggested by sponsors.

**Comment by artogahr** [comment]
Wow, good luck with the kids, I wish them a happy life!

I don't have much knowledge about Gnome extensions or AppIndicator API but I'll look into implementing it as a challenge :) If anyone has any tips to get started it would be appreciated.
